specification/voiceservices/resource-manager/Microsoft.VoiceServices/cadl/communicationsgateway.cadl

```diff
@@ -0,0 +1,196 @@
+model CommunicationsGateway is TrackedResource<CommunicationsGatewayProperties> {
+  @pattern("^[a-zA-Z0-9-]{3,24}$")
+  @key("communicationsGatewayName")
+  @segment("communicationsGateways")
+  @doc("Unique identifier for this deployment")
+  @visibility("Read")
+  @path
+  name: string;
+}
+
+@doc("The status of the current CommunicationsGateway resource.")
+enum Status {
+  @doc("The resource has been created or updated, but the CommunicationsGateway service has not yet been updated to reflect the changes.")
+  ChangePending,
+  @doc("The CommunicationsGateway service is up and running with the parameters specified in the resource.")
+  Complete,
+}
+
+@doc("The method for terminating emergency calls to the PSTN.")
+@knownValues(E911TypeValues)
+model E911Type is string {}
+enum E911TypeValues {
+  @doc("Emergency calls are not handled different from other calls")
+  Standard,
+  @doc("Emergency calls are routed directly to the ESRP")
+  DirectToEsrp
+}
+
+@doc("The voice codecs expected for communication with Teams.")
+@knownValues(TeamsCodecsValues)
+model TeamsCodecs is string {}
+enum TeamsCodecsValues {
+	PCMA,
+	PCMU,
+	G722,
+	G722_2,
+	SILK_8,
+	SILK_16
+}
+
+@doc("How this deployment connects back to the operator network")
+@knownValues(ConnectivityValues)
+model Connectivity is string {}
+enum ConnectivityValues {
+  @doc("This deployment connects to the operator network using a Public IP address, e.g. when using MAPS")
+  PublicAddress
+}
+
+@doc("The service region configuration needed for Teams Callings.")
+model ServiceRegionProperties {
+	@doc("The name of the region in which the resources needed for Teams Calling will be deployed.")
+	@visibility("read", "create")
+	name: string;
+
+	@doc("The configuration used in this region as primary, and other regions as backup.")
+	@visibility("read", "write", "create")
+	primaryRegionProperties: PrimaryRegionProperties;
+}
+
+@doc("The configuration used in this region as primary, and other regions as backup.")
+model PrimaryRegionProperties {
+	@doc("IP address to use to contact the operator network from this region")
+	@visibility("read", "write", "create")
+	operatorAddresses: string[];
+
+	@doc("IP address to use to contact the ESRP from this region")
+	@visibility("read", "write", "create")
+	esrpAddresses?: string[];
+}
+
+@doc("Available platform types.")
+@knownValues(CommunicationsPlatformValues)
+model CommunicationsPlatform is string {}
+enum CommunicationsPlatformValues {
+	OperatorConnect,
+	TeamsPhoneMobile
+}
+
+// Currently empty, with presence indicating that this feature should be enabled.
+// This is expected to include more properties in future, hence being an object.
+@doc("Details of API bridge functionality.")
+model ApiBridgeProperties {}
+
+@doc("Details of the CommunicationsGateway resource.")
+model CommunicationsGatewayProperties is ResourceProperties {
+  @doc("The current status of the deployment.")
+  @visibility("read") status?: Status;
```

Comment:
>status? [](http://example.com/codeflow?start=22&length=7)

Will the status ever be missing from this property?

Reply:
It's not required on a PUT, which is why we've made it optional. Is this incorrect?

Reply:
(We may also remove it in future if Azure Monitor etc. gives a better view of the status of the deployment, and my understanding is that it's easier to stop returning an optional field than to remove a mandatory field.)